src/libostree/ostree-repo-prune.c

```diff
@@ -153,26 +153,28 @@ _ostree_repo_prune_tmp (OstreeRepo *self,
       size_t len;
       gboolean has_sig_suffix = FALSE;
       struct dirent *dent;
+      g_autofree gchar *d_name = NULL;
 
       if (!glnx_dirfd_iterator_next_dent (&dfd_iter, &dent, cancellable, error))
         return FALSE;
       if (dent == NULL)
         break;
 
-      len = strlen (dent->d_name);
-      if (len > 4 && g_strcmp0 (dent->d_name + len - 4, ".sig") == 0)
+      d_name = g_strdup (dent->d_name);
```

Comment:
Can you add a comment that we can't reuse `dent->d_name` to increase the odds we don't regress on this?

Reply:
Done